rasa/importers/multi_project.py

```diff
@@ -162,13 +164,39 @@ def _is_in_imported_paths(self, path) -> bool:
 
     def add_import(self, path: Text) -> None:
         self._imports.append(path)
-
+    
+    def _log_merge_infos(self, domains) -> None:
+        merges = {
+            "actions": {},
+            "entities": {},
+            "forms": {},
+            "intents": {},
+            "slots": {}
+        }
+        domain_dicts = [domain.as_dict() for domain in domains]
+        for i, domain_dict in enumerate(domain_dicts):
+            domain_path = self._domain_paths[i]
+            for k, v in merges.items():
+                for e in domain_dict.get(k, []):
+                    name = e if isinstance(e, str) else next(iter(e.keys()))
+                    v[name] = v.get(name, []) + [domain_path]
```

Comment:
I suggest using separate helper methods for each of the `for` loops to make the code a bit more expressive, e.g.

```python
merge_info = self._collect_paths_for_each_property(domains)
self._log_properties_with_multiple_paths(merge_info)
```

Another thought that I had: Can use something like the approach here https://howtodoinjava.com/python/dictionary-intersection/ to simplify the code? If not, it's also ok, it was just an idea :-)

Reply:
Agreed, I restructured the code with the 2 helper methods you suggested.

Good idea about using "dictionary intersection" to find redundant properties in multiple domains, but I'm not sure it is easily applicable here (one of the reasons being that we are not looking for the properties belonging to all domains, but rather the ones belonging to at least 2 of them).